src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1291,19 +1494,47 @@ public static void ImplicitBodyNotProvided(HttpContext httpContext, string param
             [LoggerMessage(5, LogLevel.Debug, ImplicitBodyNotProvidedLogMessage, EventName = "ImplicitBodyNotProvided")]
             private static partial void ImplicitBodyNotProvided(ILogger logger, string parameterName);
 
-            public static void UnexpectedContentType(HttpContext httpContext, string? contentType, bool shouldThrow)
+            public static void UnexpectedJsonContentType(HttpContext httpContext, string? contentType, bool shouldThrow)
+            {
+                if (shouldThrow)
+                {
+                    var message = string.Format(CultureInfo.InvariantCulture, UnexpectedJsonContentTypeExceptionMessage, contentType);
+                    throw new BadHttpRequestException(message, StatusCodes.Status415UnsupportedMediaType);
+                }
+
+                UnexpectedJsonContentType(GetLogger(httpContext), contentType ?? "(none)");
+            }
+
+            [LoggerMessage(6, LogLevel.Debug, UnexpectedJsonContentTypeLogMessage, EventName = "UnexpectedContentType")]
+            private static partial void UnexpectedJsonContentType(ILogger logger, string contentType);
+
+            public static void UnexpectedFormContentType(HttpContext httpContext, string? contentType, bool shouldThrow)
             {
                 if (shouldThrow)
                 {
-                    var message = string.Format(CultureInfo.InvariantCulture, UnexpectedContentTypeExceptionMessage, contentType);
+                    var message = string.Format(CultureInfo.InvariantCulture, UnexpectedFormContentTypeExceptionMessage, contentType);
                     throw new BadHttpRequestException(message, StatusCodes.Status415UnsupportedMediaType);
                 }
 
-                UnexpectedContentType(GetLogger(httpContext), contentType ?? "(none)");
+                UnexpectedFormContentType(GetLogger(httpContext), contentType ?? "(none)");
+            }
+
+            [LoggerMessage(7, LogLevel.Debug, UnexpectedFormContentTypeLogMessage, EventName = "UnexpectedContentType")]
```

Comment:
```suggestion
            [LoggerMessage(7, LogLevel.Debug, UnexpectedFormContentTypeLogMessage, EventName = "UnexpectedNonFormContentType")]
```

We should probably give this a unique name since we gave it a unique id. Feel free to pick a better name.